src/app/clusters/bindings/BindingManager.h

```diff
@@ -118,6 +119,18 @@ class BindingManager
      */
     CHIP_ERROR NotifyBoundClusterChanged(EndpointId endpoint, ClusterId cluster, void * context);
 
+    /*
+     * Notify a cluster change to a specific bound device associated with the (endpoint, cluster) tuple.
```

Comment:
What is the use case for this?

The whole point of bindings is that whatever it is you plan to do for a specific (local endpoint, cluster) couple you do to all the bindings.  When would you only do it for _some_ bindings?

Reply:
The thought behind this change is to avoid a device waste resources on setting up a session to a bound device if the device already knows that it does not plan on interacting with that device.

As far as I know there is no way to prevent a binding being configured in the binding table, ex.in our case we have an implementation that can interact with an occupancy sensor and nothing else, so if someone sets up a binding to a light or some other random device, the application already knows that it does not make sense to interact with this device.

Reply:
In general the binding manager seems to function out of the assumption that all bindings in the binding table are valid or relevant, which for some devices might be true and for other it is not.

Reply:
Other than a controller messing up, how would a binding in the binding table not be "valid or relevant"?  And should the application really be second-guessing the controller here?

Reply:
If we keep in the pump example shown above. The implementation can only do a very specific interaction with an occupancy sensor, but there is no way for the pump to indicate this to a pump. A different version might also be able to be bound to a generic switch used to turn on and off (or some other binding that could make sense). Since the controller has no way of knowing this there is nothing preventing it to allow configuring bindings which is of no interest to a pump. Let's take an extreme example that a pump gets a binding to a light.

I would argue that it is not so much second-guessing the controller here, I would more call it sanity-checking the input it receives.

Reply:
Is it expected that a controller MUST read the ClientList of each available endpoint and only setup binding with for nodes which matches the supported device types? I do not think that is mandated in the binding cluster and will be difficult since a binding might not even have a cluster id 😞

Reply:
> Let's take an extreme example that a pump gets a binding to a light.

Why would a controller do that?

> I would more call it sanity-checking the input it receives.

How would the sanity-checking be performed, exactly?

Concretely, say the controller adds a binding from a pump to some node id.  How will the pump actually decide whether to send messages to that node?  Keep in mind that the pump may not have ACLs to read descriptor clusters on that node and whatnot.

> Is it expected that a controller MUST read the ClientList of each available endpoint and only setup binding with for nodes which matches the supported device types?

There is no normative requirement to that effect, but in practice I would expect controllers to only set up bindings that make sense for the client clusters in question, yes...

>  and will be difficult since a binding might not even have a cluster id

As in, if this is a "wildcard" binding that applies to all client clusters on the endpoint?  If it is, then the expectation is that it ... applies to all those client clusters, no?

Reply:
The functions has been removed, so this is marked as resolved